tests/components/vicare/conftest.py

```diff
@@ -35,10 +38,28 @@ def __init__(self, fixtures: list[str]) -> None:
 class MockViCareService:
     """PyVicareService mock using a json dump."""
 
-    def __init__(self, fixture: str) -> None:
+    def __init__(
+        self,
+        installation_id: str,
+        gateway_id: str,
+        device_id: str,
+        fixture: tuple[list[str], str],
+    ) -> None:
         """Initialize the mock from a json dump."""
-        self._test_data = load_json_object_fixture(fixture)
+        self._test_data = load_json_object_fixture(fixture[1])
         self.fetch_all_features = Mock(return_value=self._test_data)
+        self.roles = fixture[0]
+        self.accessor = ViCareDeviceAccessor(installation_id, gateway_id, device_id)
+
+    def hasRoles(self, requested_roles: list[str]) -> bool:
+        """Return true if requested roles are assigned."""
+        return len(requested_roles) > 0 and set(requested_roles).issubset(
+            set(self.roles)
+        )
```

Comment:
```suggestion
        return requested_roles and set(requested_roles) in self.roles
```

After `self.roles` is changed to a set

Can `requested_roles` be also a set?

Reply:
> Can requested_roles be also a set?

 I think no, as this is how it's [defined in the library](https://github.com/somm15/PyViCare/blob/63f9f7fea505fdf9a26c77c6cd0bff889abcdb05/PyViCare/PyViCareService.py#L53-L54).

Reply:
`set(requested_roles) in self.roles` did not work, tests were failing, reverted back to `issubset`. Any idea why?

Reply:
I missed something. issubset is the correct way